src/main/java/bio/terra/common/db/JdbcConfiguration.java

```diff
@@ -0,0 +1,82 @@
+package bio.terra.common.db;
+
+import java.util.Properties;
+import org.apache.commons.dbcp2.ConnectionFactory;
+import org.apache.commons.dbcp2.DriverManagerConnectionFactory;
+import org.apache.commons.dbcp2.PoolableConnection;
+import org.apache.commons.dbcp2.PoolableConnectionFactory;
+import org.apache.commons.dbcp2.PoolingDataSource;
+import org.apache.commons.lang3.builder.ToStringBuilder;
+import org.apache.commons.lang3.builder.ToStringStyle;
+import org.apache.commons.pool2.ObjectPool;
+import org.apache.commons.pool2.impl.GenericObjectPool;
+
+public class JdbcConfiguration {
+  private String uri;
+  private String username;
+  private String password;
+
+  // Not a property
+  private PoolingDataSource<PoolableConnection> dataSource;
+
+  public String getUri() {
+    return uri;
+  }
+
+  public String getUsername() {
+    return username;
+  }
+
+  public String getPassword() {
+    return password;
+  }
+
+  // NOTE: even though the setters appear unused, the Spring infrastructure uses them to populate
+  // the properties.
+  public void setUri(String uri) {
+    this.uri = uri;
+  }
+
+  public void setUsername(String username) {
+    this.username = username;
+  }
+
+  public void setPassword(String password) {
+    this.password = password;
+  }
+
+  // Main use of the configuration is this pooling data source object.
+  public PoolingDataSource<PoolableConnection> getDataSource() {
+    // Lazy allocation of the data source
+    if (dataSource == null) {
+      configureDataSource();
+    }
+    return dataSource;
+  }
+
+  private void configureDataSource() {
+    Properties props = new Properties();
+    props.setProperty("user", getUsername());
+    props.setProperty("password", getPassword());
+
```

Comment:
You might consider adding the ability to configure `apache.commons.pool2` JMX integration to avoid issues like PF-485:

https://github.com/DataBiosphere/terra-resource-janitor/pull/93

Reply:
I'm not familiar with JMX. I may wait on this.

Reply:
@ddietterich But we are using dbcp2 today, e.g. https://github.com/DataBiosphere/terra-workspace-manager/blob/dev/src/main/java/bio/terra/workspace/app/configuration/BaseDatabaseConfiguration.java

If this class would be the common base class for all apps, the easiest path to migrate is to keep using dbcp2 with jmx config for now, then switch to HikariCP in the future.